Components/Storage.js

```diff
@@ -1,49 +1,58 @@
-import AsyncStorage from '@react-native-async-storage/async-storage';
+import AsyncStorage from "@react-native-async-storage/async-storage";
+export const USERNAME_KEY = "UserName";
+export const MINFREQ_KEY = "MinFrequency";
+export const MAXFREQ_KEY = "MaxFrequency";
 
-const saveName = async () => {
+//e.g. saveName("Chesney")
+export const saveName = async (user_name) => {
   try {
-    await AsyncStorage.setItem(STORAGE_KEY, user_name)
-    alert('Data successfully saved')
+    await AsyncStorage.setItem(USERNAME_KEY, user_name);
+    alert("Data successfully saved");
   } catch (e) {
-    alert('Failed to save the data to the storage')
+    alert("Failed to save the data to the storage");
   }
-}
+};
 
-const saveLowestFreq = async () => {
+// Note: AsyncStorage prevents storing non-string values so we will store the frequencies as strings
+// e.g. saveLowestFreq(200)
+export const saveLowestFreq = async (min_freq) => {
   try {
-    await AsyncStorage.setItem(STORAGE_KEY, lowest_freq)
-    alert('Data successfully saved')
+    await AsyncStorage.setItem(MINFREQ_KEY, min_freq);
+    alert("Data successfully saved");
   } catch (e) {
-    alert('Failed to save the data to the storage')
+    console.log(e);
+    alert("Failed to save the data to the storage");
   }
-}
+};
 
-const savHighestFreq = async () => {
+export const savHighestFreq = async (max_freq) => {
   try {
-    await AsyncStorage.setItem(STORAGE_KEY, highest_freq)
-    alert('Data successfully saved')
+    await AsyncStorage.setItem(MAXFREQ_KEY, max_freq);
+    alert("Data successfully saved");
   } catch (e) {
-    alert('Failed to save the data to the storage')
+    alert("Failed to save the data to the storage");
   }
-}
+};
 
-const readData = async () => {
+export const readData = async (key) => {
   try {
-    const value = await AsyncStorage.getItem(STORAGE_KEY);
+    const value = await AsyncStorage.getItem(key);
+    return value;
 
-  if (value !== null) {
-    setInput(value);
-  }
+    // Do we need this? if key doesn't exist, it returns null?
```

Comment:
good point I don't think we do? if I end up needing it when implementing saving I can just add it back later but it should be fine without